codegenerator/cli/templates/static/codegen/src/db/EntityHistory.res

```diff
@@ -0,0 +1,325 @@
+open Table
+
+type historyFieldsGeneral<'a> = {
+  chain_id: 'a,
+  block_timestamp: 'a,
+  block_number: 'a,
+  log_index: 'a,
+}
+
+type historyFields = historyFieldsGeneral<int>
+
+type entityIdOnly = {id: string}
+let entityIdOnlySchema = S.schema(s => {id: s.matches(S.string)})
+type entityData<'entity> = Delete(entityIdOnly) | Set('entity)
+
+type historyRow<'entity> = {
+  current: historyFields,
+  previous: option<historyFields>,
+  entityData: entityData<'entity>,
+}
+
+type previousHistoryFields = historyFieldsGeneral<option<int>>
+
+//For flattening the optional previous fields into their own individual nullable fields
+let previousHistoryFieldsSchema = S.object(s => {
+  chain_id: s.field("previous_entity_history_chain_id", S.null(S.int)),
+  block_timestamp: s.field("previous_entity_history_block_timestamp", S.null(S.int)),
+  block_number: s.field("previous_entity_history_block_number", S.null(S.int)),
+  log_index: s.field("previous_entity_history_log_index", S.null(S.int)),
+})
+
+let currentHistoryFieldsSchema = S.object(s => {
+  chain_id: s.field("entity_history_chain_id", S.int),
+  block_timestamp: s.field("entity_history_block_timestamp", S.int),
+  block_number: s.field("entity_history_block_number", S.int),
+  log_index: s.field("entity_history_log_index", S.int),
+})
+
+let makeHistoryRowSchema: S.t<'entity> => S.t<historyRow<'entity>> = entitySchema => {
+  //Instantiates an entity object with all fields set to null
+  let entityWithNullFields: Js.Dict.t<unknown> = switch entitySchema->S.classify {
+  | Object({items}) =>
+    let nulldict = Js.Dict.empty()
+    items->Belt.Array.forEach(({location}) => {
+      nulldict->Js.Dict.set(location, %raw(`null`))
+    })
+    nulldict
+  | _ =>
+    Js.Exn.raiseError("Failed creating entityWithNullFields. Expected an object schema for entity")
+  }
+
+  //Gets an entity object with all fields set to null except for the id field
+  let getEntityWithNullFields = (entityId: string) => {
+    entityWithNullFields->Utils.Dict.updateImmutable("id", entityId->Utils.magic)
+  }
```

Comment:
Added this function that gets an entity with all its fields as nullable except id

Reply:
Is it for rescript-schema to pass validation? I think postgress should be fine to have the null fields omitted

Reply:
Hmmm 🤔 let me test it.